python/ray/train/_internal/worker_group.py

```diff
@@ -345,5 +345,24 @@ def add_workers(self, num_workers: int):
         for i in range(len(new_actors)):
             self.workers.append(Worker(actor=new_actors[i], metadata=metadata[i]))
 
+    def _move_workers_with_ip_to_front(self, ip):
```

Comment:
should we add a small unit test for this in `test_worker_group` with fake IPs?

Reply:
yes